### PR TITLE
Fixes #2183  - Port capnp serialization to TPRegion class

### LINKED ISSUE #2183: Port capnp serialization to TMRegion class
See parent issue: https://github.com/numenta/nupic/issues/2158

### PR DESCRIPTION
**Pull request by david-ragazzi**
Fixes: https://github.com/numenta/nupic/issues/2183

### CODE REVIEW

File: nupic/research/TP.py [hunk 1]
```diff
@@ -3280,21 +3634,75 @@ def __init__(self, tp, isSequenceSeg):
...
-  def __ne__(self, s):
-    return not self == s
+  def __eq__(self, other):
+    membersToIgnore = ['tp']
+    for k, v1 in self.__dict__.iteritems():
+      if k in membersToIgnore:
+        pass
+      else:
+        if not k in other.__dict__:
+          print 'not found: ', k
```
Comment (david-ragazzi) [inline nupic/research/TP.py]: I'll remove this `print`'s before merge it.

**Comment by scottpurdy** [comment]
Please separate the region out into another PR

**Comment by david-ragazzi** [comment]
> Please separate the region out into another PR

Done.

**Comment by scottpurdy** [comment]
Why are you including the TemporalMemoryV1Proto? If we are using separate schema since they are sufficiently different then I think it will be easier to maintain to have completely separate schema files. That way we don't have to coordinate changes between two different algorithms.

**Comment by rhyolight** [comment]
Marking as `in progress` until builds pass.

**Comment by scottpurdy** [comment]
@david-ragazzi - Getting the C++ -> Python capnp stuff working is taking a while so I will try to review your region changes now.

The TPRegion supports multiple implementations currently. The cpp/py options will need TemporalMemoryV1Proto while the tm_py option is the new implementation and will have a different schema. Should the capnp schema for the region have a [union](https://capnproto.org/language.html#unions) or [AnyPointer](https://capnproto.org/language.html#dynamically-typed-fields) field to support the two different cases?

**Pull request closed without merging**